.github/workflows/dependencies-and-licenses.yml

```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 #v5.3.0
         with:
-          go-version: '~1.23'
+          go-version: '~1.24'
```

Comment:
Use go.mod to get go version version from go.mod. 
Take a look: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Reply:
Like we could have `go-version-file: 'path/to/go.mod'` and then we wouldn't need to update it separately here?

Reply:
> Like we could have `go-version-file: 'path/to/go.mod'` and then we wouldn't need to update it separately here?

exactley

Reply:
The main goal of this PR is updating the Go version to a new version, and everything beyond is already stretching that.
Since this `go-version-file` is a different approach to what we have now, I don't see this as a blocker for this PR.
However, I will create a followup-PR introducing this.